contracts/turnstile.sol

```diff
@@ -18,28 +20,20 @@ contract Turnstile {
         uint64 id
     );
 
+    constructor() {
+        tokenID = 0;
+    }
+
     // register the smart contract to an existing CSR nft
     function register(uint64 id) public {
-        emit UpdateCSREvent(msg.sender, id);
+        tokenID++;
+        emit Attach(msg.sender, id);
     }
 
     // register and mint a new CSR nft that will be transferred to the to address entered
     function register(address to) public {
-        emit RegisterCSREvent(msg.sender, to);
-    }
-
-    // deploys a smart contract and registers the newly deployed smart contract to
-    // an existing CSR nft
-    function deploy(bytes[] memory contractByteCode, uint64 nft_id) public {
-        // createThisSmartContract(contractByteCode)
-        // emit UpdateCSREvent(newSmartContractAddress, nft_id)
-    }
-
-    // deploys a smart contract and registers the newly deployed smart contract
-    // to a new CSR nft
-    function deploy(bytes[] memory contractByteCode) public {
-        // createThisSmartContract(contractByteCode)
-        // emit RegisterCSREvent(newSmartContractAddress, msg.sender)
+        tokenID++;
+        emit Register(msg.sender, to, tokenID);
```

Comment:
These methods were made only for testing purpose? In this case shld we implement a sample distribute fees method on the test Turnstile contract / methods to interact in x/csr?

Reply:
will be done today